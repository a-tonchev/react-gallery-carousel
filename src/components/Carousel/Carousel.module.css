.carousel {
  position: relative;
  width: 100%;
  height: 100%;
  user-select: none;
  display: flex;
  background-color: #fff;
  flex-flow: column;
  align-items: stretch;
  box-sizing: border-box;
}

.maxCarousel {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 2000;
  width: 100%;
  height: 100%;
  background-color: #fff;
  transform: scale(1);
  transform-origin: top center;
  display: flex;
  flex-flow: column;
  align-items: stretch;
}

.carouselInner {
  width: 100%;
  height: 100%;
  overflow: hidden;
  position: relative;
  flex: 0 1 auto;
}

.slidesContainer {
  width: 100%;
  height: 100%;
  overflow: hidden;
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  touch-action: auto;
  flex: 0 1 auto;
}

.gallery {
  background-color: #000000;
}

[data-is-not-keyboard-user='true'] *:focus {
  outline: none;
}
