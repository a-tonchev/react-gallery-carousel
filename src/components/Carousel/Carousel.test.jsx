import React from 'react';
import renderer from 'react-test-renderer';
import Carousel from './index';
import { defaultProps } from './props';
const { describe, it, expect } = global;

const images = [900, 800, 700, 600, 500].map((size) => ({
  src: `https://placedog.net/${size}/${size}`
}));

const imageIDs = Array(6)
  .fill(1)
  .map((_, i) => i + 1);
const images2 = imageIDs.map((imageID) => {
  return {
    src: `https://placedog.net/8${imageID}0/6${imageID}0?id=${imageID}`,
    alt: `Dog No. ${imageID}. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated.`,
    thumbnail: `https://placedog.net/8${imageID}/6${imageID}?id=${imageID}`
  };
});

// snapshot testing
describe('Carousel', () => {
  it('is truthy', () => {
    expect(Carousel).toBeTruthy();
  });

  global.IntersectionObserver = function () {
    return {
      observe: () => {},
      disconnect: () => {}
    };
  };

  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: (query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: () => {},
      removeEventListener: () => {}
    })
  });

  it('renders the same as snapshot', () => {
    const component = renderer.create(
      <Carousel {...defaultProps} images={images} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can be maximized and renders the same as snapshot', () => {
    const component2 = renderer.create(
      <Carousel {...defaultProps} images={images2} isMaximized />
    );
    const tree2 = component2.toJSON();
    expect(tree2).toMatchSnapshot();
  });
});
