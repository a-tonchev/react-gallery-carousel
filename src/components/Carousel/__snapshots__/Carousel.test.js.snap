// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel can be maximized and renders the same as snapshot 1`] = `
Array [
  <div
    className="carousel gallery"
  />,
  <div
    className="maxCarousel gallery"
    style={Object {}}
  />,
  <div
    className="maxCarousel gallery"
    data-is-not-keyboard-user="true"
    style={Object {}}
  >
    <div
      className="carouselInner"
    >
      <div
        className="slidesContainer"
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUpCapture={[Function]}
        tabIndex={0}
      >
        <ul
          className="slides hasImages"
        >
          <li
            aria-live="polite"
            className="slide"
            role="presentation"
          >
            <figure
              aria-label="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="figure"
            >
              <img
                alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="image"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                srcSet={null}
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
              <img
                alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="thumbnail"
                onError={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
            </figure>
          </li>
          <li
            aria-live={null}
            className="slide"
            role="presentation"
          >
            <figure
              aria-label="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="figure"
            >
              <img
                alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="image"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                srcSet={null}
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
              <img
                alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="thumbnail"
                onError={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
            </figure>
          </li>
          <li
            aria-live={null}
            className="slide"
            role="presentation"
          >
            <figure
              aria-label="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="figure"
            >
              <img
                alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="image"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                srcSet={null}
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
              <img
                alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="thumbnail"
                onError={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
            </figure>
          </li>
          <li
            aria-live={null}
            className="slide"
            role="presentation"
          >
            <figure
              aria-label="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="figure"
            >
              <img
                alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="image"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                srcSet={null}
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
              <img
                alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="thumbnail"
                onError={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
            </figure>
          </li>
          <li
            aria-live={null}
            className="slide"
            role="presentation"
          >
            <figure
              aria-label="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="figure"
            >
              <img
                alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="image"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                srcSet={null}
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
              <img
                alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="thumbnail"
                onError={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
            </figure>
          </li>
          <li
            aria-live={null}
            className="slide"
            role="presentation"
          >
            <figure
              aria-label="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
              className="figure"
            >
              <img
                alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="image"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                srcSet={null}
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
              <img
                alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
                className="thumbnail"
                onError={[Function]}
                src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
                style={
                  Object {
                    "objectFit": "contain",
                  }
                }
              />
            </figure>
          </li>
        </ul>
      </div>
      <div
        className="widgetWrapper topLeft"
      >
        <button
          aria-disabled={false}
          aria-label="Start Autoplay"
          className="button"
          onClick={[Function]}
          type="button"
        >
          <div
            className="iconWrapper square"
          >
            <svg
              className="icon"
              height="30"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 30 30"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m1 1v28l28-14z"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        className="widgetWrapper textWrapper topCenter"
        tabIndex="-1"
      >
        <span
          className="text"
        >
          1
           / 
          6
        </span>
      </div>
      <div
        className="widgetWrapper topRight"
      >
        <button
          aria-disabled={false}
          aria-label="Minimize Slides"
          className="button"
          onClick={[Function]}
          type="button"
        >
          <div
            className="iconWrapper square"
          >
            <svg
              className="icon"
              height="30"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 30 30"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 1v5h-5v5h10v-10zm13 0v10h10v-5h-5v-5zm-18 18v5h5v5h5v-10zm18 0v10h5v-5h5v-5z"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        className="widgetWrapper centerLeft"
      >
        <button
          aria-disabled={false}
          aria-label="Go to Previous Slide"
          className="button"
          onClick={[Function]}
          type="button"
        >
          <div
            className="iconWrapper rectangle"
          >
            <svg
              className="icon"
              height="40"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 20 40"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m19 1v37l-18-20z"
              />
            </svg>
          </div>
        </button>
      </div>
      <div
        className="widgetWrapper centerRight"
      >
        <button
          aria-disabled={false}
          aria-label="Go to Next Slide"
          className="button"
          onClick={[Function]}
          type="button"
        >
          <div
            className="iconWrapper rectangle"
          >
            <svg
              className="icon"
              height="40"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 20 40"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m1 1v38l18-19z"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
    <div
      className="thumbnailsWrapper"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUpCapture={[Function]}
      onWheel={[Function]}
      style={
        Object {
          "flexBasis": undefined,
        }
      }
    >
      <ul
        className="thumbnails"
      >
        <li
          className="thumbnail currentThumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          style={
            Object {
              "maxWidth": undefined,
              "minWidth": undefined,
              "width": undefined,
            }
          }
          tabIndex={0}
        >
          <img
            alt="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 1. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            onError={[Function]}
            src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          style={
            Object {
              "maxWidth": undefined,
              "minWidth": undefined,
              "width": undefined,
            }
          }
          tabIndex={0}
        >
          <img
            alt="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 2. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            onError={[Function]}
            src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          style={
            Object {
              "maxWidth": undefined,
              "minWidth": undefined,
              "width": undefined,
            }
          }
          tabIndex={0}
        >
          <img
            alt="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 3. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            onError={[Function]}
            src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          style={
            Object {
              "maxWidth": undefined,
              "minWidth": undefined,
              "width": undefined,
            }
          }
          tabIndex={0}
        >
          <img
            alt="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 4. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            onError={[Function]}
            src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          style={
            Object {
              "maxWidth": undefined,
              "minWidth": undefined,
              "width": undefined,
            }
          }
          tabIndex={0}
        >
          <img
            alt="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 5. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            onError={[Function]}
            src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
          />
        </li>
        <li
          className="thumbnail"
          onMouseUpCapture={[Function]}
          role="button"
          style={
            Object {
              "maxWidth": undefined,
              "minWidth": undefined,
              "width": undefined,
            }
          }
          tabIndex={0}
        >
          <img
            alt="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            aria-label="Dog No. 6. Dogs are domesticated mammals, not natural wild animals. They were originally bred from wolves. They have been bred by humans for a long time, and were the first animals ever to be domesticated."
            className="image"
            onError={[Function]}
            src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
          />
        </li>
      </ul>
    </div>
  </div>,
]
`;

exports[`Carousel renders the same as snapshot 1`] = `
<div
  className="carousel gallery"
  data-is-not-keyboard-user="true"
>
  <div
    className="carouselInner"
  >
    <div
      className="slidesContainer"
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUpCapture={[Function]}
      tabIndex={0}
    >
      <ul
        className="slides hasImages"
      >
        <li
          aria-live="polite"
          className="slide"
          role="presentation"
        >
          <figure
            aria-label={null}
            className="figure"
          >
            <img
              alt={null}
              className="image"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              srcSet={null}
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
            <img
              alt={null}
              className="thumbnail hidden"
              onError={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
          </figure>
        </li>
        <li
          aria-live={null}
          className="slide"
          role="presentation"
        >
          <figure
            aria-label={null}
            className="figure"
          >
            <img
              alt={null}
              className="image"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              srcSet={null}
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
            <img
              alt={null}
              className="thumbnail hidden"
              onError={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
          </figure>
        </li>
        <li
          aria-live={null}
          className="slide"
          role="presentation"
        >
          <figure
            aria-label={null}
            className="figure"
          >
            <img
              alt={null}
              className="image"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              srcSet={null}
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
            <img
              alt={null}
              className="thumbnail hidden"
              onError={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
          </figure>
        </li>
        <li
          aria-live={null}
          className="slide"
          role="presentation"
        >
          <figure
            aria-label={null}
            className="figure"
          >
            <img
              alt={null}
              className="image"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              srcSet={null}
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
            <img
              alt={null}
              className="thumbnail hidden"
              onError={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
          </figure>
        </li>
        <li
          aria-live={null}
          className="slide"
          role="presentation"
        >
          <figure
            aria-label={null}
            className="figure"
          >
            <img
              alt={null}
              className="image"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              srcSet={null}
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
            <img
              alt={null}
              className="thumbnail hidden"
              onError={[Function]}
              src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
              style={
                Object {
                  "objectFit": null,
                }
              }
            />
          </figure>
        </li>
      </ul>
    </div>
    <div
      className="widgetWrapper topLeft"
    >
      <button
        aria-disabled={false}
        aria-label="Start Autoplay"
        className="button"
        onClick={[Function]}
        type="button"
      >
        <div
          className="iconWrapper square"
        >
          <svg
            className="icon"
            height="30"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1 1v28l28-14z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      className="widgetWrapper textWrapper topCenter"
      tabIndex="-1"
    >
      <span
        className="text"
      >
        1
         / 
        5
      </span>
    </div>
    <div
      className="widgetWrapper topRight"
    >
      <button
        aria-disabled={false}
        aria-label="Maximize Slides"
        className="button"
        onClick={[Function]}
        type="button"
      >
        <div
          className="iconWrapper square"
        >
          <svg
            className="icon"
            height="30"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 30 30"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1 1v10h5v-5h5v-5zm18 0v5h5v5h5v-10zm-18 18v10h10v-5h-5v-5zm18 5v5h10v-10h-5v5z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      className="widgetWrapper centerLeft"
    >
      <button
        aria-disabled={false}
        aria-label="Go to Previous Slide"
        className="button"
        onClick={[Function]}
        type="button"
      >
        <div
          className="iconWrapper rectangle"
        >
          <svg
            className="icon"
            height="40"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 20 40"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m19 1v37l-18-20z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      className="widgetWrapper centerRight"
    >
      <button
        aria-disabled={false}
        aria-label="Go to Next Slide"
        className="button"
        onClick={[Function]}
        type="button"
      >
        <div
          className="iconWrapper rectangle"
        >
          <svg
            className="icon"
            height="40"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 20 40"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1 1v38l18-19z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div
    className="thumbnailsWrapper"
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUpCapture={[Function]}
    onWheel={[Function]}
    style={
      Object {
        "flexBasis": undefined,
      }
    }
  >
    <ul
      className="thumbnails"
    >
      <li
        className="thumbnail currentThumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        style={
          Object {
            "maxWidth": undefined,
            "minWidth": undefined,
            "width": undefined,
          }
        }
        tabIndex={0}
      >
        <img
          alt={null}
          aria-label={null}
          className="image"
          onError={[Function]}
          src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        style={
          Object {
            "maxWidth": undefined,
            "minWidth": undefined,
            "width": undefined,
          }
        }
        tabIndex={0}
      >
        <img
          alt={null}
          aria-label={null}
          className="image"
          onError={[Function]}
          src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        style={
          Object {
            "maxWidth": undefined,
            "minWidth": undefined,
            "width": undefined,
          }
        }
        tabIndex={0}
      >
        <img
          alt={null}
          aria-label={null}
          className="image"
          onError={[Function]}
          src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        style={
          Object {
            "maxWidth": undefined,
            "minWidth": undefined,
            "width": undefined,
          }
        }
        tabIndex={0}
      >
        <img
          alt={null}
          aria-label={null}
          className="image"
          onError={[Function]}
          src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
        />
      </li>
      <li
        className="thumbnail"
        onMouseUpCapture={[Function]}
        role="button"
        style={
          Object {
            "maxWidth": undefined,
            "minWidth": undefined,
            "width": undefined,
          }
        }
        tabIndex={0}
      >
        <img
          alt={null}
          aria-label={null}
          className="image"
          onError={[Function]}
          src="data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAAAAAA6fptVAAAACklEQVR4nGNiAAAABgADNjd8qAAAAABJRU5ErkJggg=="
        />
      </li>
    </ul>
  </div>
</div>
`;
